src/libvirtApi/domain.js

```diff
@@ -138,9 +123,32 @@ export function domainAttachDisk({
     busType,
     serial,
 }) {
-    const xmlDesc = getDiskXML(type, file, device, poolName, volumeName, format, target, cacheMode, shareable, busType, serial);
-
-    return domainAttachDevice({ connectionName, vmId, permanent, hotplug, xmlDesc });
+    const options = { err: "message" };
+    if (connectionName === "system")
+        options.superuser = "try";
+    let update = "";
+    if (hotplug)
+        update = "--update";
+    let define = "--define";
+    if (hotplug && !permanent)
+        define = "--no-define";
+    let source = "";
+    if (type === 'file')
+        source = `,source.file=${file}`;
+    else
+        source = `,source.pool=${poolName},source.volume=${volumeName}`;
+    let driverType = "";
+    if (format && ['qcow2', 'raw'].includes(format))
+        driverType = `,driver.type=${format}`;
+    let serialNum = "";
+    if (serial)
+        serialNum = `,serial=${serial}`;
+    const shareableOption = shareable ? "yes" : "no";
+
+    return cockpit.script(
```

Comment:
Please use cockpit.spawn() for this, pass this as a proper array, and avoid the shell.